### DISCUSSION TITLE
RQL - How to load array field?

### DISCUSSION THREAD

**Discussion by damccull** [ravendb/ravendb#15014] [Q&A]
Hello,
I'm doing all this in the studio.

I've got the following document...
```json
{
    "TestingLocation": "TestingLocation/BobsBurgers",
    "PointsOfContact": [
        {
            "PointOfContact": "PointOfContact/BobBelcher",
            "Type": "Tenant"
        },
        {
            "PointOfContact": "PointOfContact/LindaBelcher",
            "Type": "Tenant"
        }
    ],
    "@metadata": {
        "@collection": "TestingOrder"
    }
}
```
...and I want to make a javascript projection in RQL that shows the first two points of contact last name. Here is an example PointOfContact:
```json
{
    "LastName": "Belcher",
    "FirstName": "Bob",
    "Phone": "123-456-7890",
    "Email": "bob@bobsburgers.com",
    "@metadata": {
        "@collection": "PointOfContact"
    }
}
```
Below is the query I'm trying:
```rql
from TestingOrder as o
load
  o.TestingLocation as t,
  o.PointsOfContact as poc,
  poc[0].PointOfContact as poc1,
  poc[1].PointOfContact as poc2
select {
    TestLocation: t.Name,
    PoC1: poc1.LastName,
    PoC2: poc[1].PointOfContact.LastName,
}
```
Both ways of trying this, I always just get `null`. How do I access arrays in documents in RQL?

**Comment by arekpalinski** [comment]
Hi,

You can use `load()` directly in JavaScript projection code:

```
from TestingOrder as o
select {
    TestLocation: load(o.TestingLocation).Name,
    PoC1: load(o.PointsOfContact[0].PointOfContact).LastName,
    PoC2: load(o.PointsOfContact[1].PointOfContact).LastName,
}
```

**Reply by damccull** [reply]
Awesome, thanks!

Now, I got this working in a query but how does it work in an index? I tried modifying it as:
```rql
from o in docs.TestingOrder
select new
{
    TestLocation = load(o.TestingLocation).Name,
    PoC1 = load(o.PointsOfContact[0].PointOfContact).LastName,
    PoC2 = load(o.PointsOfContact[1].PointOfContact).LastName,
}
```
but I get the error `error CS0103: The name 'load' does not exist in the current context` for each use of `load()`.

**Reply by arekpalinski** [reply]
For indexing purposes we have `LoadDocument()` method:

https://ravendb.net/docs/article-page/5.4/csharp/indexes/indexing-related-documents

**Reply by damccull** [reply]
Sorry for the work, but am I misunderstanding how indexes work or just doing this wrong? Here's the Map index I'm trying to create as a test:
```rql
from o in docs.TestingOrder
select new
{
    TestLoc = (this.LoadDocument(o.TestingLocation,"TestingLocation")).Name,
}
```
However, when querying the index, I only see the exact fields defined on the document. My TestLoc field isn't there, but a bunch of others are.

If I also include this 'Reduce', I get something like what I want. However, I am not trying to create aggregates with this index. I'm not counting anything. More like, I'm trying to create View like you'd have in MSSQL.

```rql
from r in results
group r by r.TestLoc into g
select new
{
      TestLoc = "Test"
}
```

Is this not something Map-only indexes can do, or am I just doing it wrong?

**Reply by ppekrol** [reply]
By default fields in index are used only for querying capabilities (not for data retrieval), you can store the values https://ravendb.net/docs/article-page/5.4/csharp/indexes/storing-data-in-index and then perform a projection https://ravendb.net/docs/article-page/5.4/csharp/indexes/querying/projections to get them back

I think it would be also good for you to understand the query flow: https://ravendb.net/docs/article-page/5.4/csharp/indexes/querying/basics#query-flow
